js/eventloop/eventloop.go

```diff
@@ -55,11 +55,62 @@ func (e *EventLoop) wakeup() {
 	}
 }
 
-// RegisterCallback register that a callback will be invoked on the loop, preventing it from returning/finishing.
-// The returned function, upon invocation, will queue its argument and wakeup the loop if needed.
-// If the eventLoop has since stopped, it will not be executed.
-// This function *must* be called from within running on the event loop, but its result can be called from anywhere.
-func (e *EventLoop) RegisterCallback() func(func() error) {
+// RegisterCallback is deprecated, see RegisterPendingCallback.
+//
+// Deprecated: due to the confusing name, we renamed this method to
+// RegisterPendingCallback.
+func (e *EventLoop) RegisterCallback() func(func() error) { // TODO: remove
+	return e.RegisterPendingCallback()
+}
+
+// RegisterPendingCallback signals to the event loop that you are going to do
+// some asynchronous work off the main thread and that you may need to execute
+// some code back on the main thread when you are done. So, once you call this
+// method, the event loop will wait for you to finish and give it the callback it
+// needs to run back on the main thread before it can end the whole iteration.
+//
+// RegisterPendingCallback() *must* be called from the main runtime thread, but
+// its result addToMainThreadQueue() is thread-safe and can be called from any
+// goroutine. addToMainThreadQueue() ensures that its callback parameter is
+// added to the VU runtime's tasks queue, to be executed on the main runtime
+// thread eventually, when the VU is done with the other tasks before it. Unless
+// the whole event loop has been stopped, invoking addToMainThreadQueue() will
+// queue its argument and "wake up" the loop (if it was idle, but not stopped).
+//
+// Keep in mind that once you call RegisterPendingCallback(), you *must* also
+// call addToMainThreadQueue() exactly once, even if don't actually need to run
+// any code on the main thread. If that's the case, you can pass an empty no-op
+// callback to it, but you must call it! The event loop will wait for the
+// addToMainThreadQueue() invocation and the k6 iteration won't finish and will
+// be stuck until the VU itself has been stopped (e.g. because the whole test or
+// scenario has ended).
+//
+// A common pattern for async work is something like this:
+//
+//    func doAsyncWork() *goja.Promise {
+//        addToMainThreadQueue := vu.RegisterPendingCallback()
+//        p, resolve, reject := runtime.NewPromise()
+//
+//        go func() {
+//            result, err := doTheActualAsyncWork()
```

Comment:
```suggestion
//            result, err := doTheActualAsyncWork() // the actual work here
```

Reply:
isn't the function name enough? why repeat it as a comment?

Reply:
I would argue it will help with the explanation and drawing the attention to that specific line.

Reply:
Resolved by https://github.com/grafana/k6/pull/2615/commits/23efb9583bfd5280ecac3fcbd333e7a713d752ed